plugins/reboot/reboot.go

```diff
@@ -50,24 +102,51 @@ func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlug
 		return nil, engines.ErrContractViolation
 	}
 
+	pl.runningTasks.RLock()
 	return taskPlugin{
 		TaskPluginBase: plugins.TaskPluginBase{},
-		reboot:         p.Reboot,
+		plugin:         pl,
+		rebootWhenDone: p.RebootWhenDone,
 	}, nil
 }
 
 func (tp taskPlugin) Dispose() error {
-	if tp.reboot {
-		if err := reboot(); err != nil {
-			return engines.NewInternalError(err.Error())
-		}
+	tp.plugin.runningTasks.RUnlock()
+
+	// Note: in a multi-task environment, this will not reboot immediately, but
+	// will wait until all tasks are complete.
+	if tp.rebootWhenDone {
+		tp.plugin.environment.Log.Warning("Task specified reboot after completion; scheduling reboot")
+		tp.plugin.reboot()
 	}
 
 	return nil
 }
 
 func (pluginProvider) NewPlugin(options plugins.PluginOptions) (plugins.Plugin, error) {
-	return plugin{}, nil
+	var c config
+	if err := schematypes.MustMap(configSchema, options.Config, &c); err != nil {
+		return nil, engines.ErrContractViolation
+	}
+
+	plugin := &plugin{
+		PluginBase:   plugins.PluginBase{},
+		environment:  options.Environment,
+		runningTasks: sync.RWMutex{},
+	}
+
+	if c.RebootAfter > 0 {
```

Comment:
Where would that conditional go?  *Something* needs to execute at that time.  I could make a timer that runs that check every hour but that would be a waste of time for the first 95 invocations. 

And we need to avoid race conditions between checking time since startup and checking for idleness.  The RWLock actually models that quite cleanly.